storage/jellyfish_merkle/src/mock_tree_store.rs

```diff
@@ -19,6 +19,31 @@ impl TreeReader for MockTreeStore {
     fn get_node_option(&self, node_key: &NodeKey) -> Result<Option<Node>> {
         Ok(self.0.read().unwrap().0.get(node_key).cloned())
     }
+
+    fn get_rightmost_leaf(&self) -> Result<Option<(NodeKey, LeafNode)>> {
+        let locked = self.0.read().unwrap();
+        let mut node_key_and_node: Option<(NodeKey, LeafNode)> = None;
+
+        for (key, value) in locked.0.iter() {
```

Comment:
(despite it being a mock) shall we use a btreemap here to avoid the full scan? (I'm not sure how frequently this will be possibly used in tests in the future.)

Reply:
oh i see `NodeKey` is not in that order. Not worth it to implement an alternative order just for the mock.